AppCenterDataStorage/AppCenterDataStorage/MSAbstractDocument.m

```
@@ -0,0 +1,32 @@
+#import "MSAbstractDocument.h"
+#import "MSAppCenterInternal.h"
+#import "MSDataStoreInternal.h"
+
+@implementation MSAbstractDocument
+
+- (instancetype)initFromDictionary:(NSDictionary *)__unused dictionary {
+  
+  // This method needs to be implemented by the super class.
+  @try {
```

You should just log an error I don't think you need to raise the exception.